commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpClientFactory.java

```diff
@@ -99,8 +133,45 @@ public C createConnection(final String hostname, final int port, char[] username
                 password = ANON_CHAR_ARRAY;
             }
 
+            Integer connectionTimeout = null;
+            try {
+                connectionTimeout = Integer.parseInt(timeout);
+            } catch (NumberFormatException nfe) {
+                connectionTimeout = 5000;
```

Comment:
If 'timeout' parsing fails we are setting a default value to 'connectionTimeout'?  Ideally shouldn't we handle the exception with a meaningful log to inform that the parsing failed?